File: invoiceJavaBackend/src/main/java/com/example/invoiceJavaBackend/dto/InvoiceDetailsDTO.java
package com.example.invoiceJavaBackend.dto;

import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotEmpty;

public class InvoiceDetailsDTO {
    
    @Valid
    private InvoiceDTO invoice;

    @Valid
    private ContractorDTO contractor;
    
    @Valid
    @NotEmpty
    private List<ItemDTO> items;

    public InvoiceDetailsDTO() {
        
    }

    public List<ItemDTO> getItems() {
        return items;
    }

    public void setItems(List<ItemDTO> items) {
        this.items = items;
    }

    public ContractorDTO getContractor() {
        return contractor;
    }

    public void setContractor(ContractorDTO contractor) {
        this.contractor = contractor;
    }

    public InvoiceDTO getInvoice() {
        return invoice;
    }

    public void setInvoice(InvoiceDTO invoice) {
        this.invoice = invoice;
    }

}

File: InvoiceSharpWPFApp/InvoiceSharpWPFApp/Dto/InvoiceDetailsDTO.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InvoiceSharpWPFApp.Dto
{
    public class InvoiceDetailsDTO
    {
        public InvoiceDetailsDTO()
        {
            invoice = new InvoiceDTO();
            contractor = new ContractorDTO();
            items = new ObservableCollection<ItemDTO>();
        }

        public InvoiceDTO invoice { get; set; }
        public ContractorDTO contractor { get; set; }
        public ObservableCollection<ItemDTO> items { get; set; }
    }
}
